src/Extension/BetterNavigatorExtension.php

```diff
@@ -76,14 +95,65 @@ public function BetterNavigator() {
             ]);
 
             // Merge with page data, send to template and render
-            $bNData = new ArrayData($bNData);
-            $page = $this->owner->customise(['BetterNavigator' => $bNData]);
-            return $page->renderWith('BetterNavigator\\BetterNavigator');
```

Comment:
I haven't tested this but it looks like the controller scope has been lost here. Do the various links that include `$AbsoluteLink` in the template still work? Aside from that, if devs opt to make use of the include hooks such as `<% include BetterNavigator\BetterNavigatorExtraDevTools %>` they would probably be expecting the template to be scoped to the current controller.

Reply:
Yeah, tough one to explain here, but I think I've got it sorted in my latest commit. Basically, what was happening is that due to the prerendering in `onBeforeActionHandler`, values were being cached in SSViewer that were returning unexpected results when the final render happened. Specifically, `$Me` was always coming back as `ArrayData`, which was killing my project.

I had a quick look and wondered why the controller was even scoped at all, ran a few tests, which worked, because `<% base_tag %>` was providing the `$AbsoluteLink` fallback, and all was good. But I see now why you need the controller involved.

Interestingly, `$arrayData->customise($controller)->renderWith(..)` worked fine, but when the controller was the parent scope, the value of `$Me` was just getting clobbered.

In any event, I've refactored to remove prerendering. All it does now is pull in the requirements in the before handler. You lose the template level `<% require %>`, but the whole thrust of this PR is to get rid of the reliance on templates, so I don't see how that will be a problem. Custom includes should render the same.

Reply:
I wonder if `$Me` was getting clobbered because you've taken a slightly different approach. When I customised the controller previously I sandboxed the array data in a BetterNavigator variable `customise(['BetterNavigator' => $bNData])`. You've merged the array data straight in (`$this->owner->customise($navigator)`), perhaps overriding some properties of the controller?

Reply:
Yeah, could be. I think the newest approach makes more sense, from my perspective.